setup.py

```diff
@@ -0,0 +1,43 @@
+from setuptools import setup, find_packages
+
+setup(
+    name='itemloaders',
+    version='0.1',
+    url='---',
+    project_urls={
+        'Documentation': '---',
+        'Source': '---',
+        'Tracker': '---',
+    },
+    description='---',
+    long_description='---',
+    author='---',
+    maintainer='---',
+    maintainer_email='---',
+    license='BSD',
+    packages=find_packages(exclude=('tests', 'tests.*')),
+    include_package_data=True,
+    zip_safe=False,
+    classifiers=[
+        'Framework :: Scrapy',
+        'Development Status :: 5 - Production/Stable',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: BSD License',
+        'Operating System :: OS Independent',
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.5',
+        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: Implementation :: CPython',
+        'Programming Language :: Python :: Implementation :: PyPy',
+    ],
+    python_requires='>=3.5',
+    install_requires=[
+        'w3lib==1.21.0',
+        'parsel==1.5.2',
+        'jmespath==0.9.5'
```

Comment:
install_requires shouldn't fix package versions; here we can only exclude versions which are known not to work. E.g. w3lib >= 1.21.0 is ok, but w3lib==1.21.0 is not.

Reply:
We're should also keep a single source of truth. There's a requirements.txt file with the same dependencies.

Reply:
A common rule is to use requirements.txt to ping dependencies which are known to work, and install_requires to exclude dependencies which are known not to work. Though I'm not sure why requirements.txt file is needed here.

Reply:
Ok, I'll rename `requirements.txt` to `requirements-dev.txt`

Reply:
Regarding keeping it in two places.
The options I see are:

* `requirements-dev.txt` all dependencies
* `setup.py` package dependencies
* Only `pip install -r requirements-dev.txt`

The options I see are:

* `requirements-dev.txt` only dev dependencies (pytest, flake)
* `setup.py` package dependencies
*  `pip install -r requirements-dev.txt`
* `pip install .`

Is there any other option?

Reply:
An approach which we usually follow:

* dependencies which are required to use the package are in install_requires
* dependencies which are required to run tests or other checks (flake8, mypy, sphinx) are in tox.ini file. Sometimes tox.ini can use requirements file (e.g. requirements-tests.txt or tests/requirements.txt), to make it easier to organize. 

We should be running these checks automatically on CI; for local development users shouldn't worry about installing them, they should use tox.